### PR TITLE
Feature/4179/refresh org confirmation

### PR DESCRIPTION
**Pull request by garyluu**
**Description**
Add a confirmation dialog before refreshing an org in my-workflows. If there's any additional text to add to the dialog, now is the time to bring it up.

**Issue**
For dockstore/dockstore#4179

### CODE REVIEW

File: cypress/integration/group2/myworkflows.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { Repository } from '../../../src/app/shared/openapi/model/repository';
 import { goToTab, isActiveTab, resetDB, setTokenUserViewPort } from '../../support/commands';
 
 describe('Dockstore my workflows', () => {
-  resetDB();
+  // resetDB();
```
Comment (garyluu) [inline cypress/integration/group2/myworkflows.ts:20]: eeek

File: src/app/shared/refresh-organization/refresh-organization.component.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
   type="button"
   class="mr-1"
   data-cy="refreshOrganization"
-  (click)="refreshOrganization()"
+  (click)="openConfirmationDialog()"
```
Comment (coverbeck) [inline src/app/shared/refresh-organization/refresh-organization.component.html:23]: Not that we have very many of those types of orgs, and I don't know if it's easy to tell, but ideally we wouldn't have the dialog pop up with text talking about GitHub apps if it's a Bitbucket or Gitlab org.

File: src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts [hunk 3]
```diff
@@ -54,6 +58,24 @@ export class RefreshWorkflowOrganizationComponent extends RefreshOrganizationCom
     this.isRefreshing$ = this.alertQuery.showInfo$;
   }
 
+  openConfirmationDialog() {
+    const confirmationDialogData: ConfirmationDialogData = {
+      title: 'Refresh Organization',
+      message: `Are you sure you wish to refresh the organization?
+                It has no effect on GitHub Apps workflows.`,
```
Comment (coverbeck) [inline src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:65]: "GitHub Apps workflows" -> I think we use it as a shorthand, but I'm not sure it will click right away for end-users.

Perhaps, "It has no effect on workflows already being synchronized via the Dockstore GitHub App"?

File: src/app/myworkflows/my-workflow/my-workflow.component.ts [hunk 4]
```diff
@@ -239,6 +239,8 @@ export class MyWorkflowComponent extends MyEntry implements OnInit {
  * @template T
  */
 export interface OrgWorkflowObject<T> extends OrgEntryObject<T> {
+  // Swagger doesn't return an enm for this. Currently, some values present are:
```
Comment (denis-yuen) [inline src/app/myworkflows/my-workflow/my-workflow.component.ts:242]: typo

File: src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts [hunk 5]
```diff
@@ -54,6 +58,28 @@ export class RefreshWorkflowOrganizationComponent extends RefreshOrganizationCom
...
 
+  ngOnChanges() {
+    this.isGitHubOrg = this.orgWorkflowObject.sourceControl === 'github.com';
+  }
+
+  openConfirmationDialog() {
+    const confirmationDialogData: ConfirmationDialogData = {
+      title: 'Refresh Organization',
+      message: `Are you sure you wish to refresh the organization?
+                It has no effect on workflows already being synchronized via the Dockstore GitHub App.`,
```
Comment (denis-yuen) [inline src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:69]: I forget the term for this, but add what it does do as opposed to what it doesn't do (or do both). 

Something like "Are you sure you wish to refresh the organization? This will sequentially refresh workflows that have been added through the Dockstore UI and will have no effect on workflows already being synchronized via the Dockstore GitHub App" except maybe better

Comment (garyluu) [reply src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:69]: I think it's getting a little too wordy. It'll also affect workflows added through CLI too.

**Review by coverbeck — APPROVED** [review]

File: src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts (hunk 5 above)
Comment (denis-yuen) [reply src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:69]: :thinking:

in the meantime, you should probably take-out/change the tooltip too if you have not already 
Screenshot from 2021-11-04 17-36-59

Comment (garyluu) [reply src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:68]: Removed the tooltip as per slack discussion.

"Are you sure you wish to refresh the organization? This will sequentially refresh workflows that have been added through the Dockstore website/CLI and will have no effect on workflows already being synchronized via the Dockstore GitHub App"

is too wordy imo, doesn't provide any additional useful information and could possibly be missing other sources. Others should probably chime in (and provide reasoning)

Comment (garyluu) [reply src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:68]: How about: "This will sequentially refresh all workflows in the organization except for those already being synchronized via the Dockstore GitHub App."?

Comment (denis-yuen) [reply src/app/workflow/refresh-workflow-organization/refresh-workflow-organization.component.ts:68]: That works for me

**Review by denis-yuen — COMMENTED** [review]
proposed text works for me

**Review by denis-yuen — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Review by david4096 — APPROVED** [review]

**Review by wshands — APPROVED** [review]
